File: ido_sdk_flutter/ido_sdk/protocol_sdk/plugins/native_channel/lib/pigeon_impl/get_file_info_impl.dart
import 'package:native_channel/pigeon_generate/get_file_info.g.dart';

abstract class GetFileInfo {

  factory GetFileInfo() => GetFileInfoImpl();

  /// 读取文件信息
  /// {"createSeconds":0,"changeSeconds":0}
  Future<Map?> readFileInfo(String path);

}

class GetFileInfoImpl extends ApiGetFileInfo implements GetFileInfo {

  static GetFileInfoImpl? _instance;
  factory GetFileInfoImpl() => _instance ??= GetFileInfoImpl._internal();
  GetFileInfoImpl._internal();

  final _getFileInfo = ApiGetFileInfo();

  @override
  Future<Map?> readFileInfo(String path) {
     return _getFileInfo.readFileInfo(path);
  }

}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/plugins/native_channel/android/src/main/kotlin/com/idosmart/native_channel/GetFileInfoImpl.kt
package com.idosmart.native_channel

import com.idosmart.native_channel.pigeon_generate.api_get_file_info.ApiGetFileInfo
import java.io.File

class GetFileInfoImpl: ApiGetFileInfo {

    companion object {
        @Volatile
        private var instance: GetFileInfoImpl? = null
        fun instance(): GetFileInfoImpl {
            return instance ?: synchronized(this) {
                instance ?: GetFileInfoImpl().also { instance = it }
            }
        }
    }

    override fun readFileInfo(path: String, callback: (Result<Map<Any, Any?>?>) -> Unit) {
//        println("readFileInfo === $path")
       val file = File(path)
        if (file.exists() && file.isFile) {
            val map = GetFileInfoTool.getFileAttributes(path)
            NativeChannelPlugin.instance().androidLog("android get file info == $map")
            callback(Result.success(map) as Result<Map<Any, Any?>?>)
        }else {
            callback(Result.success(mutableMapOf(
                Pair("createSeconds", 0),
                Pair("changeSeconds", 0)
            )))
        }
    }

}